### PR TITLE
Shield powerup

### PR DESCRIPTION
**Pull request by hooger**
I created a distinct commit out of the shield powerup changes. I made some modifications to the original PR, now the animation flashes on every hit.

### CODE REVIEW

**Comment by ooxi** [comment]
Sorry for the slow feedback, currently looking at this :-)

File: src/game/lifeforms/Player.cpp [hunk 1]
```diff
@@ -44,6 +46,7 @@ Player::Player(float x, float y, Sprite *legsSprite, Sprite *deathSprite,
 	m_arms = NULL;
 	m_body = new DynamicObject(x, y, legsSprite);
 	m_body->Frame = 8;
+	m_shield = new DynamicObject(x, y, shieldSprite);
```
Comment (ooxi) [inline src/game/lifeforms/Player.cpp]: Shouldn't `m_shield` be `delete`d in the destructor?

**Comment by ooxi** [comment]
_Loving_ the new shield mechanics! I think they trigger only rarely, compared to the other powerups. However, this is caused by the broken roulette system and can be fixed in a separate commit.

I'm going to merge this :-) maybe we should release a new version as soon as the control presets have landed?

**Comment by ooxi** [comment]
Merged your commit as e10b13de8efda3ed3517bd83fe23f452481f0dc7. Thank you very much for this great addon!

I have reset the po-updates, since there is quite a high line noise caused by whitespace changes :-)

File: src/game/lifeforms/Player.cpp (hunk 1 above)
Comment (hooger) [reply src/game/lifeforms/Player.cpp]: You are right, I totally forgot that. I try to add it later that day.

**Comment by hooger** [comment]
> Edit: Are you using a different mail address on GitHub then you are using in your commit? GitHub does not pick up the connection between you hooger Account and the mail address. If you want I can amend the commit to the mail address [hooger@users.noreply.github.com](mailto:hooger@users.noreply.github.com) instead of the currently used one.

My mail can be different, I'm really new to git best practices... 


> _Loving_ the new shield mechanics! I think they trigger only rarely, compared to the other powerups. However, this is caused by the broken roulette system and can be fixed in a separate commit.

I agree, that it triggers too rarely, I couldn't show it to a friend I wanted to show off to. :D I wanted to open an issue to figure out how to rebalance the drop of the powerups. They show up too rarely until you pick Looting, which becomes a must, this way. I also had some other gameplay mechanics idea, where should we discuss it?

**Pull request closed without merging**